### DISCUSSION TITLE
Make Query Widget detachable

### DISCUSSION THREAD

**Discussion by Assad-Ashraf** [TanStack/query#4695] [Ideas]
Hi, I have worked on Redux Toolkit. You people are taking the React Lovers to heavenly env by this package. I was wondering if we can make the widget detachable from window?
I am a MERN stack developer and can contribute to implement this idea too.

**Comment by TkDodo** [comment]
are you talking about the `devtools` ? What would `detachable` mean exactly?

**Reply by Assad-Ashraf** [reply]
yes i am asking about devtools, was thinking if they can come out of the tab and become a separate tab to play with. so that one can see all the browser's tab as their application and devtools are in separate tab/window.

**Reply by TkDodo** [reply]
I don't think we can easily make it as its own browser tab because browser tabs can't easily communicate with each other.

**Reply by DamianOsipiuk** [reply]
This is circling back to the old idea of converting `Devtools` from a `react component` to a `browser extension`.
It could be tailored to handle all of tanstack packages via a single browser extension.

And yes, making it work is a mess, but it could be based on existing work 👀 Vue Devtools, React Devtools etc.

**Reply by Assad-Ashraf** [reply]
Yup making an extension for React Query can easily help do this, @TkDodo if you people are looking for a hand to start this idea. I can start working on it as a side project. It's a need anyways, because inspect tab like view of RQ tools makes the actual app interaction loose.

**Reply by TkDodo** [reply]
sure, feel free to start working on it on your own terms and then we can circle back once you have something 👍
